### DISCUSSION TITLE
Can buildpacks be used for OS Image?

### DISCUSSION THREAD

**Discussion by alpana17** [buildpacks/community#256] [Q&A]
If I take source code of say Ubuntu 22.
Now, I try to build it.
I want to create a layer system.
Like one layer represents base Image. Next layer represents OS hardening. Next represents logging mechanism inside OS. etc.
Can buildpacks be used here?

**Comment by natalieparellano** [comment]
I think you would be better off using a Dockerfile here. Buildpacks run in userspace and can only write to specific directories, so they likely wouldn't be able to perform all of the functions you are describing.
